### PR TITLE
[#606] Enable use of deploy key for publishing

### LINKED ISSUE #606: You may want to use a dummy account for the GitHub token
I feel it is better to use a dummy account to generate the token and add the dummy account as a collaborator to run dashboard update/deploy in the Travis build.

Relevant document to update: https://github.com/reposense/RepoSense/blob/release/docs/PublishingGuide.md

### PR DESCRIPTION
**Pull request by yong24s**
Fixes #606 

```
In the publishing guide, instructions were given to generate personal
access token to grant write-access to RepoSense for publishing.

Despite ease of setup, personal access token may not be as secure as it
can access all other repositories owned by the user.

Let's also support deploy key for enhanced security, which enables user
to grant access to only a single repository.
```

PR for deploy script at https://github.com/reposense/publish-RepoSense/pull/5

### CODE REVIEW

File: docs/PublishingGuide.md [hunk 1]
```diff
@@ -23,6 +22,20 @@
...
 Try accessing your site again when a green tick appears beside your fork.
 
+### Granting write-access to RepoSense for publishing
+We recommmend use of [personal access token](https://github.blog/2013-05-16-personal-api-tokens/) for ease of setup and [deploy key](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys) for enhanced security.
+
+##### If you wish to use `personal access token`:
+1. Follow this [guide](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) and give only `public_repo` permission <br/>
+1. **Copy** the token for later use <br/>
+
+##### If you wish to use `deploy key`:
```
Comment (damithc) [inline docs/PublishingGuide.md]: Do these instructions work for Windows users?

Comment (yong24s) [reply docs/PublishingGuide.md]: Not with vanilla cmd. They would have to use online tools.

Comment (yong24s) [reply docs/PublishingGuide.md]: Not sure, if it's a good idea to link users to online tools. Since these are mean to be created securely 😅.

Comment (damithc) [reply docs/PublishingGuide.md]: I see. can recommend alternatives for Windows users?

File: docs/PublishingGuide.md [hunk 2]
```diff
@@ -23,6 +22,20 @@
...
+1. Use `ssh-keygen` to create a public/private key pair without a passphrase <br/>
+i.e. `ssh-keygen -t ecdsa -b 521 -f id_reposense -q -N ""`
+1. Go to the [deploy key settings](../../../../publish-RepoSense/settings/keys) of your publish-RepoSense fork and create a new deploy key with the contents of `id_reposense.pub`
+1. **Copy** the base64 encoded content of the private key for later use <br/>
+i.e. `cat id_reposense | base64 -w 0`
```
Comment (damithc) [inline docs/PublishingGuide.md]: This 'later use' is something mentioned later in our documentation?

Comment (yong24s) [reply docs/PublishingGuide.md]: Yes, *later* in the setup instructions.

In Step 7:
7. Under **Environment Variables**, name a variable as `GITHUB_TOKEN` or `GITHUB_DEPLOY_KEY` depending on your earlier choice and paste the content that was copied earlier to its value field; then click **Add** <br/>

Comment (damithc) [reply docs/PublishingGuide.md]: Got it 👍

**Pull request merged**